Tools/Sources/needletools/Utilities/Version.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  Copyright (c) 2018. Uber Technologies
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// Errors that can occur when parsing the properly formatted version
+/// `String`.
+enum VersionErrors: Error {
+    /// If the original value is not in the format of `major.minor.patch`,
+    /// where all components are numbers only.
+    case invalidFormat
+}
+
+/// The utilities of package version.
+class Version: Comparable {
+
+    /// The major part.
+    let major: Int
+    /// The minor part.
+    let minor: Int
+    /// The patch part.
+    let patch: Int
+    /// The string representation of this version.
+    lazy var stringValue = "v\(major).\(minor).\(patch)"
+
+    /// Initializer.
+    ///
+    /// - parameter string: The `String` to parse a version from.
+    /// - throws: `VersionErrors.invalidFormat` if the given `String` value
+    /// is not in the format for `major.minor.patch` where all parts are
+    /// digits.
+    init(string: String) throws {
+        let parts = string.split(separator: ".")
+            .map { (substring: Substring) -> String in
+                var string = String(substring)
+                string.removeAll { (c: Character) -> Bool in
+                    let cSet = CharacterSet(charactersIn: String(c))
+                    return CharacterSet.decimalDigits.intersection(cSet).isEmpty
+                }
+                return string
```

Comment:
nit: if we `return Int(string)!` here, then it'll be a bit more clear why the force-cast should be mostly safe.

Reply:
Good point. I'll update.